exonum/src/storage/proof_map_index/key.rs

```diff
@@ -191,10 +191,10 @@ impl DBKey {
         self.to == (KEY_SIZE * 8) as u16
     }
 
-    // TODO: terrible hack, try to remove this
-    pub fn to_vec(&self) -> Vec<u8> {
-        let mut buffer = vec![0u8; DB_KEY_SIZE as usize];
-        self.write(&mut buffer);
+    /// Represents `DBKey` as byte array and returns it
+    pub fn as_bytes(&self) -> Box<[u8]> {
```

Comment:
As for me, that isn't very important, but `Box<[u8]>` is slightly "stricter": it has no dynamic size/capacity. :smile: